src/models/attributes.ts

```diff
@@ -193,7 +198,12 @@ class AttributeCollection extends Attribute {
     }
     const objs = [];
     for (const objJSON of json) {
-      const obj = new this.itemClass(_parent.connection, objJSON);
+      let obj;
+      if (this.itemClass.prototype instanceof RestfulModel) {
```

Comment:
I like this pattern, but I think it can be improved a bit by having `RestfulModel`'s constructor accept the connection object as the second prop instead of the first. That way, you don't need these `if` checks and can just try to instantiate it like this.
```
obj = new this.itemClass(objJSON, _parent.connection);
```
If `itemClass` is of type `Model`, it would just ignore the second param in its constructor. The downside of doing this is that we'd have to implement some conditional logic if `Model` ever requires a second param in the future.

Reply:
That's actually a really clean idea! My only concern is that the objJSON parameter (json) is optional for both `RestfulModel` and `Model` so I can't make it the first parameter on `RestfulModel` unless I make `connection: NylasConnection` optional (which is not ideal since we need this parameter).

Reply:
Oh I didn't realize they were optional, let's just leave it as is then